libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -208,15 +208,15 @@ proc send*(p: PubSubPeer, msg: RPCMsg, anonymize: bool) =
   # or malicious data on the wire - in particular, re-encoding protects against
   # some forms of valid but redundantly encoded protobufs with unknown or
   # duplicated fields
+  var mm = msg # hooks can modify the message
```

Comment:
this was intentionally moved into the `if`, to avoid copying the message when it's not necessary (it's a waste, and mostly we don't h ave send observers)

Reply:
if there are no send observers, it would in fact be better to do the object -> protobuf conversion once for all peers - alternatively, don't give send observers per-peer powers

Reply:
it breaks metrics, I can rework it to not copy.

Reply:
yes, this is a hotspot in gossipsub